### ISSUE TITLE
Metac should emit global symbol definitions even if they don't have occurrences

### ISSUE THREAD

**Issue by olafurpg** [scalameta/scalameta#1504]
While testing the new hover implementation in metals with semanticdb v3.7, I hit on 
```
java.util.NoSuchElementException: org.langmeta.lsp.Hover#
```
which can be explained by the fact that `@JsonCodec class Hover` has a macro annotation

https://github.com/scalameta/metals/blob/fcac1cc31fedf0a2dd8a1c723cc97c59a398edca/lsp4s/src/main/scala/org/langmeta/lsp/Commands.scala#L201

The annotation is messing up with positions causing there to be no matching occurrence for the definition. This breaks both goto-definition as well as symbol lookups.

- Metac should emit SymbolInformation for ALL definitions, regardless if they have a matching occurrence or not
- Metac should emit a SymbolOccurrence for global definitions even if they have no matching position in g.Tree, this should accommodate macro annotations that mess up positions

In a nutshell, metac should syntactically emit symbol occurrences for global definitions like tags does.

**Comment by olafurpg** [comment]
Hit on another likely case where this might be causing problems

```scala
final case class Complex[@sp(Float, Double) T](real: T, imag: T)
// java.util.NoSuchElementException: spire.math.Complex#[T]
```
